### PR TITLE
allow for pinning of ycsb load/workload pods

### PR DESCRIPTION
**Pull request by aakarshg**
This commit will facilitate pinning of the workload/load pods
to a specific nodes, and also in the process moves the pod/job
definition into ansible templates

### CODE REVIEW

File: roles/load-ycsb/templates/workload.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          value: "{{ycsb.operationcount}}"
+        - name: uuid
+          value: "{{uuid}}"
+        imagePullPolicy: Always
+        command: ["/bin/sh", "-c"]
+        args: ["python /opt/snafu/run_snafu.py --tool ycsb -r 1 -l -d {{ ycsb.driver }} -u {{ uuid }} --user {{ test_user | default(\"ripsaw\") }} -w {{ ycsb_workload }} -x \"{{ ycsb.options_load }}\""]
+        volumeMounts:
+          - name: config-volume
+            mountPath: "/tmp/ycsb"
+{% if ycsb.pin is sameas true %}
```
Comment (rsevilla87) [inline roles/load-ycsb/templates/workload.yml.j2:43]: If ycsb is not defined, it will fail. Do we want to require this parameter? If so, we should update tests.

File: roles/ycsb-bench/templates/workload.yml.j2 [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        - name: uuid
+          value: "{{uuid}}"
+        imagePullPolicy: Always
+        command: ["/bin/sh", "-c"]
+        args: ["python /opt/snafu/run_snafu.py --tool ycsb -r 1 -d {{ ycsb.driver }} -w {{ ycsb_workload }} -u {{ uuid }} --user {{ test_user | default(\"ripsaw\") }} -x \"{{ ycsb.options_run }}\""]
+        volumeMounts:
+          - name: config-volume
+            mountPath: "/tmp/ycsb"
+      restartPolicy: Never
+{% if ycsb.pin is sameas true %}
```
Comment (rsevilla87) [inline roles/ycsb-bench/templates/workload.yml.j2:44]: Same as above.

Comment (rsevilla87) [inline roles/ycsb-bench/templates/workload.yml.j2:44]: Same as above

File: roles/load-ycsb/templates/workload.yml.j2 (hunk 1 above)
Comment (rsevilla87) [inline roles/load-ycsb/templates/workload.yml.j2:43]: If ycsb.pin is not defined, it will fail. Do we want to require this parameter?

File: roles/load-ycsb/templates/workload.yml.j2 [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            value: {{ycsb_workload}}
+          - name: num_records
+            value: {{ycsb.recordcount}}
+          - name: num_operations
+            value: {{ycsb.operationcount}}
+          - name: uuid
+            value: {{uuid}}
+          imagePullPolicy: Always
+          command: ["/bin/sh", "-c"]
+          args: ["python /opt/snafu/run_snafu.py --tool ycsb -r 1 -l -d {{ ycsb.driver }} -u {{ uuid }} --user {{ test_user | default("ripsaw") }} -w {{ ycsb_workload }} -x '{{ ycsb.options_load }}'"]
```
Comment (rsevilla87) [inline roles/load-ycsb/templates/workload.yml.j2:40]: According to CI, looks like there's a problem with the quotes
```
yaml.scanner.ScannerError: while scanning a plain scalar\\n  in \\\"<string>\\\", line 40, column 168:\\n     ... w workloada -x '-p mongodb.url=\\\"mongodb://mongo-0.mongo/ycsb?\\\"'\\\"]\\n                                         ^\\nfound unexpected ':'\\n  in \\\"<string>\\\", line 40, column 175:\\n     ... oada -x '-p mongodb.url=\\\"mongodb://mongo-0.mongo/ycsb?\\\"'\
```

Comment (aakarshg) [reply roles/load-ycsb/templates/workload.yml.j2:40]: yep this was the error i was talking about on the last wednesday call, i'm not quite sure what would work here..

File: docs/ycsb.md [hunk 4]
```diff
@@ -30,6 +30,8 @@ spec:
         - workloadb
       options_load: '-p mongodb.url="mongodb://mongo/ycsb?"' #passed as is to ycsb when loading database
       options_run: '-p mongodb.url="mongodb://mongo/ycsb?" -threads 10 -target 100'
```
Comment (rsevilla87) [inline docs/ycsb.md:32]:
```suggestion
      options_load: "-p mongodb.url='mongodb://mongo/ycsb?'" #passed as is to ycsb when loading database
      options_run: "-p mongodb.url='mongodb://mongo/ycsb?'" -threads 10 -target 100'
```

Comment (rsevilla87) [inline docs/ycsb.md:32]: @aakarshg  Swapping single quotes by double quotes like here, at tests/test_crs/valid_ycsb-mongo.yaml should work

```suggestion
      options_load: "-p mongodb.url='mongodb://mongo/ycsb?'"
      options_run: "-p mongodb.url='mongodb://mongo/ycsb?' -threads 10 -target 100"
```

File: docs/ycsb.md [hunk 5]
```diff
@@ -28,8 +28,10 @@ spec:
       workloads:
         - workloada
         - workloadb
-      options_load: '-p mongodb.url="mongodb://mongo/ycsb?"' #passed as is to ycsb when loading database
-      options_run: '-p mongodb.url="mongodb://mongo/ycsb?" -threads 10 -target 100'
+      options_load: "-p mongodb.url='mongodb://mongo-0.mongo/ycsb?'" #passed as is to ycsb when loading database
```
Comment (dry923) [inline docs/ycsb.md:31]: @aakarshg Any reason for the change from '->" ? It's not an issue just curious why the change.

File: roles/load-ycsb/templates/workload.yml.j2 [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        - name: workload
+          value: "{{ ycsb_workload }}"
+        - name: num_records
+          value: "{{ ycsb.recordcount }}"
+        - name: num_operations
+          value: "{{ ycsb.operationcount }}"
+        - name: uuid
+          value: "{{ uuid }}"
+        command: ["/bin/sh", "-c"]
+        args: ["python /opt/snafu/run_snafu.py --tool ycsb -r 1 -l -d {{ ycsb.driver }} -u {{ uuid }} --user {{ test_user | default("ripsaw") }} -w {{ ycsb_workload }} -x '{{ ycsb.options_load }}'"]
```
Comment (dry923) [inline roles/load-ycsb/templates/workload.yml.j2:39]: @aakarshg swapping "->' in the template should fix your "ing issue that you had earlier as well as require no change to existing cr files. I do the same thing in https://github.com/cloud-bulldozer/ripsaw/pull/339

```
        args: ['python /opt/snafu/run_snafu.py --tool ycsb -r 1 -l -d {{ workload_args.driver }} -u {{ uuid }} --user {{ test_user | default("ripsaw") }} -w {{ ycsb_workload_load }} -x "{{ workload_args.options_load }}"']
```

File: docs/ycsb.md (hunk 5 above)
Comment (aakarshg) [reply docs/ycsb.md:31]: the solution to the issue was to either do this here in the cr, or do that in j2 file, @rsevilla87 suggested changes with cr worked so i tried the same and it worked. but given it looks like changes to j2 file also work, i'll go with that so we cna merge this, and not impact crs.
